kafka/client.go

```diff
@@ -180,9 +179,9 @@ func (c *client) Producer() *kafka.Writer {
 		Addr:                   kafka.TCP(c.config.Kafka.Brokers...),
 		Balancer:               &kafka.Hash{},
 		BatchSize:              c.config.Kafka.ProducerBatchSize,
-		BatchBytes:             math.MaxInt,
```

Comment:
batch bytes was setting with statically as maxInt, now it's getting default value from `ApplyDefaults()` func as 10485760.

Reply:
@halilkocaoz I think we should set it math.MaxInt too because we may sent two request while calculating the batches. 
Example: 
If we set ProducerBatchBytes as 1MB and if our batch size become 1.1MB flush will be triggered, in segmentio side this 1.1MB request will be sent as two request. We need to disable segmentio batch algorithm I think what do you think?

Reply:
> @halilkocaoz I think we should set it math.MaxInt too because we may sent more request while calculating the batches. Example: If we set ProducerBatchBytes as 1MB and if our batch size become 1.1MB flush will be triggered, in segmentio side this 1.1MB request will be sent as two request(because segmentio batch size 1MB). We need to disable segmentio batch algorithm I think what do you think?

@mhmtszr, Is [kafka-go/writer.go.writeMessages](https://github.com/segmentio/kafka-go/blob/f2d9e087266352d276b3cb951b4c668c114b8b75/writer.go#L1026) func you mean with 'two request'?

Reply:
> > @halilkocaoz I think we should set it math.MaxInt too because we may sent more request while calculating the batches. Example: If we set ProducerBatchBytes as 1MB and if our batch size become 1.1MB flush will be triggered, in segmentio side this 1.1MB request will be sent as two request(because segmentio batch size 1MB). We need to disable segmentio batch algorithm I think what do you think?
> 
> @mhmtszr, Is [kafka-go/writer.go.writeMessages](https://github.com/segmentio/kafka-go/blob/f2d9e087266352d276b3cb951b4c668c114b8b75/writer.go#L1026) func you mean with 'two request'?

size control and flushing are costly works and segmentation side already does. it would be better for me to let you decide, also refactoring that is complication for me.

Reply:
@halilkocaoz yes that writeMessages functions will seperate our one request because of segmentio batch algorithm. We can keep it max I think.

Reply:
@halilkocaoz if you set max int for ProducerBatchBytes we won't be able to batch messages(only with ticker) we just need to set max int for segmentio client BatchBytes paramater. Default ProducerBatchBytes should be still 10485760